packages/flutter/test/widgets/physical_model_test.dart

```diff
@@ -69,4 +70,336 @@ void main() {
       skip: !Platform.isLinux,
     );
   });
+
+  group('PhysicalModelLayer checks elevation', () {
+    Future<void> _testStackChildren(
+      WidgetTester tester,
+      List<Widget> children,
+      int expectedErrorCount,
+      [bool enableCheck = true]
+    ) async {
+      if (enableCheck) {
+        debugCheckElevationsEnabled = true;
+      } else {
+        assert(expectedErrorCount == 0, 'Cannot expect errors if check is disabled.');
+      }
+      debugDisableShadows = false;
+      int count = 0;
+      final Function oldOnError = FlutterError.onError;
+      FlutterError.onError = (FlutterErrorDetails details) {
+        count++;
+      };
+      await tester.pumpWidget(Directionality(
+        textDirection: TextDirection.ltr,
+        child: Stack(
+            children: children,
+          ),
+        ),
+      );
+      FlutterError.onError = oldOnError;
+      expect(count, expectedErrorCount);
+      if (enableCheck) {
+        debugCheckElevationsEnabled = false;
+      }
+      debugDisableShadows = true;
+    }
+
+    testWidgets('entirely overlapping, direct child', (WidgetTester tester) async {
```

Comment:
Future me who may have to debug these in the future wishes there were diagrams on these too